extensions/wikia/UserRenameTool/process/ProcessGlobal.class.php

```diff
@@ -0,0 +1,597 @@
+			// for confirmation before running the process
+			return false;
+		}
+
+		return empty( $this->errors );
+	}
+
+	/**
+	 * Make sure the process will not be stopped in the middle of renaming the user
+	 */
+	private function setExecutionLimits() {
+		set_time_limit( self::MAX_EXECUTION_TIME );
+		ignore_user_abort( true );
+		ini_set( "max_execution_time", self::MAX_EXECUTION_TIME );
+	}
+
+	/**
+	 * Get the old username, sanitizing the username input
+	 *
+	 * @return String
+	 */
+	private function getOldUserName() {
+		// Sanitize input data
+		$oldTitle = \Title::makeTitle( NS_USER, trim( str_replace( '_', ' ', $this->requestData->oldUsername ) ) );
+		$oldUserName = is_object( $oldTitle ) ? $oldTitle->getText() : '';
+		$this->phalanxTest( $oldUserName );
+
+		return $oldUserName;
+	}
+
+	/**
+	 * Get the new username, sanitizing the username input
+	 *
+	 * @return String
+	 */
+	private function getNewUserName() {
+		global $wgContLang;
+
+		// Force uppercase of new username, otherwise wikis with wgCapitalLinks=false can create lc usernames
+		$newTitle = \Title::makeTitleSafe( NS_USER, $wgContLang->ucfirst( $this->requestData->newUsername ) );
+		$newUserName = is_object( $newTitle ) ? $newTitle->getText() : '';
+		$this->antiSpoofTest( $newUserName );
+		$this->phalanxTest( $newUserName );
+
+		return $newUserName;
+	}
+
+	private function areUserNamesValid( $oldUserName, $newUserName ) {
+		if ( empty( $oldUserName ) ) {
+			$this->addError(
+				wfMessage( 'userrenametool-errorinvalid', $this->requestData->oldUsername )
+					->inContentLanguage()
+					->text()
+			);
+			return false;
+		}
+
+		if ( empty( $newUserName ) ) {
+			$this->addError(
+				wfMessage( 'userrenametool-errorinvalidnew', $this->requestData->newUsername )
+					->inContentLanguage()
+					->text()
+			);
+			return false;
+		}
+
+		if ( $this->areUserNamesIdentical( $oldUserName, $newUserName ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private function setNamesAndIds( $uid, \User $newUser, \User $oldUser ) {
+		$fakeUid = 0;
+
+		// If new user name does exist (we have a special case - repeating rename process)
+		if ( $this->newUserExists( $newUser ) ) {
+			$oldTitle = \Title::makeTitleSafe( NS_USER, $oldUser->getName() );
+			$newUserNameAlreadyExistsOut = $this->handleIfNewUserNameAlreadyExists(
+				$uid,
+				$newUser,
+				$oldUser,
+				$oldTitle
+			);
+
+			if ( !empty( $newUserNameAlreadyExistsOut ) ) {
+				list( $oldUser, $uid, $fakeUid ) = $newUserNameAlreadyExistsOut;
+			}
+		}
+
+		$this->oldUsername = $oldUser->getName();
+		$this->newUsername = $newUser->getName();
+		$this->userId = (int) $uid;
+		$this->fakeUserId = $fakeUid;
+	}
+
+	protected function newUserExists( \User $newUser ) {
+		$dbr = wfGetDB( DB_SLAVE, [], \F::app()->wg->ExternalSharedDB );
+
+		$table = \UserRenameToolHelper::getCentralUserTable();
+		$id = $dbr->selectField(
+			$table,
+			'user_id',
+			[ 'user_name' => $newUser->getName() ],
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "user_name" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)